app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,32 @@
+# responsible for creating and deleting user's favorite tracks
+# depends on asset_id
+# user must be logged_in
+# when favorite track is created it triggers favorites_count counter cache on asset
+class FavoritesController < ApplicationController
+  before_action :find_user
+  before_action :find_asset
+  before_action :require_login
+  before_action :find_favorite_track, only: :destroy
+
+  def create
+    raise ArgumentError unless @asset
+
+    @user.tracks.favorites.create(asset_id: @asset.id)
+    head :ok
+  end
+
+  def destroy
+    @favorite_track.destroy
```

Comment:
Does this make sense? Probably not....

```suggestion
    @favorite_track.soft_delete
```